lib/Mongo/MongoCommandCursor.php

```diff
@@ -112,4 +124,37 @@ public function __sleep()
     {
         return ['command'] + parent::__sleep();
     }
+
+    /**
+     * @see https://github.com/mongodb/mongo-php-driver-legacy/blob/1.6.14/db.c#L51
+     * @return bool
+     */
+    private function supportsReadPreference()
+    {
+        if ($this->command === []) {
+            return false;
```

Comment:
Good point about the object - I'll have to check that. I've only used arrays, but it's worth adding a test against objects as well 👍